pkg/sql/plan/function/builtin/binary/timediff.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binary
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/timediff"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func TimeDiff[T timediff.DiffT](vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	//input vectors
+	firstVector := vectors[0]
+	secondVector := vectors[1]
+	firstValues := vector.MustTCols[T](firstVector)
+	secondValues := vector.MustTCols[T](secondVector)
+	resultType := types.T_varchar.ToType()
+
+	if firstVector.IsScalarNull() || secondVector.IsScalarNull() {
+		return proc.AllocScalarNullVector(resultType), nil
+	}
+
+	vectorLen := len(firstValues)
+	if firstVector.IsScalar() {
+		vectorLen = len(secondValues)
+	}
+
+	resultVector, err := proc.AllocVectorOfRows(resultType, 0, nil)
```

Comment:
do you want to alloc vector of vectorLen rows here?

Reply:
no，there should alloc a Vector with length equals to 0， and the func vector.AppendString(resultVector, rs, proc.Mp()) will make the vector‘s length equals to vectorLen.( because of the return type is types.varchar and we use string there)